### DISCUSSION TITLE
[svelte template] `.svelte` modules are not recognized in ts files: "Cannot find module './App.svelte' or its corresponding type declarations"

### DISCUSSION THREAD

**Discussion by bnn1** [wxt-dev/wxt#301] [Q&A]
hi guys. i created a wxt project with svelte template. typescript doesn't recognize `.svelte` file import in ts files
![image](https://github.com/wxt-dev/wxt/assets/55663639/3437963d-283d-43b5-a7a6-ae3908338e5e)

can't reproduce this in other environments (i.e. [stackblitz](https://stackblitz.com/~/github.com/bnn1/wxt-bug-report-svelte-template) has no issues). any ideas what might be wrong with my setup?

**Comment by aklinker1** [comment]
Make sure the [`wxt-env.d.ts`](https://github.com/wxt-dev/wxt/blob/main/templates%2Fsvelte%2Fwxt-env.d.ts) file is included in your typescript project. It's what adds the types for `.svelte` files.

Can you share your `tsconfig.json`? Maybe you change the `include` or `exclude` fields, and now TS is ignoring that file.

**Reply by bnn1** [reply]
It's a default project bootstraped with `pnpx wxt@latest init wxt-proj`, with no changes. Interestingly, I don't have this problem on another PC 🤔
I'm using insiders version of vscode, maybe that's the case. I'll update vscode and let you know

**Reply by aklinker1** [reply]
TLDR: Don't forget to run `pnpm i`

---

I just tried it out. I saw the `Cannot find module './App.svelte'` error until I installed dependencies. Then sometimes I had to restart the language server (`cmd+P` or `ctrl+P` > Typescript: Restart TS Server) for it to pick up on the svelte types in the `node_modules` folder.

https://github.com/wxt-dev/wxt/assets/10101283/860e7512-47f9-4a1d-bba9-3e299601d521

**Reply by bnn1** [reply]
I did install dependencies. On the other PC I have no problem with recognizing `.svelte` imports, but `.svg` are not recognized. Updating vscode did nothing in this case. And also, when I click 'go to type defenitions' in the `d.ts` file, I get "not found" message

https://github.com/wxt-dev/wxt/assets/55663639/d73e4c27-e22c-4bd9-8360-7c9a73dd1e8e

**Reply by aklinker1** [reply]
I have specific checks to make sure this works in CI, and it's working for me and your other computers, so I think you're right, it's got to be something weird with your setup, like you said... There's a couple more things to check, but I'm running out of ideas lol.

- Can you run `pnpm check`? This run's Svelte's TS compiler and checks for type errors. If that also reports errors, then it's a a problem with your entire setup. If it doesn't report any errors, it's just a problem with your editor
- What version of node are you using?
- In  the video, you don't run an install command, but there is a `node_modules` folder, which makes me think you're regenerating the project in an existing folder? You could try deleting your `node_modules` folder and running `pnpm wxt clean`, then running `pnpm i` again.
- What version of typescript is installed (`pnpm why typescript`) vs in use by VS Code? Try this if they're different versions: https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-the-workspace-version-of-typescript

**Reply by bnn1** [reply]
I actually did run `pnpm install` in the video, but I accidentally cut this moment out when editing the video, sorry for the confusion 😅. It's super weird; I just created a new WXT project and there are no errors 🤔. I guess we can call it one of the quirks of VSCode. Or TypeScript. Or maybe the simulated universe glitched. Or perhaps I'm still in a coma and this is all just a dream.

**Reply by bnn1** [reply]
I just tried to create a project using `npm`, `yarn` and `bun` - no issues, no module resolution errors. I try to create a project with `pnpm` - bam, resolution errors. I guess it has something to do with how `pnpm` resolves and installs packages. Will try to investigate further a bit later.

**Reply by aklinker1** [reply]
Huh, weird. Maybe you have a different version of PNPM installed on that computer and it has a bug in it?

**Reply by bnn1** [reply]
I uninstalled `pnpm` and installed it via `npm i -g`. Still get the `cannot find module` error. Tried using the stable build of VSCode - same error.
`pnpm check`: 
![image](https://github.com/wxt-dev/wxt/assets/55663639/0f692953-8804-4613-b273-d3267268dbf3)
I'm on the LTS version of Node (v20.10)
`typescript 5.3.3`

Maybe [this](https://github.com/pnpm/pnpm/issues/5716) is related

**Reply by aklinker1** [reply]
If uninstalling and reinstalling didn;t work, did you try deleting your virtual store like mentioned in this comment? https://github.com/pnpm/pnpm/issues/5716#issuecomment-1333363011

```sh
# Check the store path. For me:  /Users/aklinker1/Library/pnpm/store/v3
pnpm store path

# Remove the whole pnpm folder
rm -rf "/Users/aklinker1/Library/pnpm"
```

That seems a likely solution to me. You could also run `pnpm store status` before deleting it to see if there are any modified dependencies that might be causing this issue.
